src/ngx_http_lua_headers.c

```diff
@@ -28,6 +28,12 @@ static int ngx_http_lua_ngx_req_header_set(lua_State *L);
 static int ngx_http_lua_ngx_resp_get_headers(lua_State *L);
 
 
+#if defined(nginx_version) && nginx_version >= 1011011
+static ngx_buf_t  **busy_bufs_ptrs = NULL;
+static ngx_int_t    prealloc_nbusy = 0;
```

Comment:
This is a bad idea to use global C variables since we have to be super careful for global variable handling during HUP reload and etc. Better use the `main_conf_t` instead to avoid all such headaches.